clients/src/test/java/org/apache/kafka/common/message/MessageTest.java

```diff
@@ -49,50 +49,127 @@
+        testMessageRoundTrips(new MetadataRequestData().setTopics(
+                Arrays.asList(new MetadataRequestData.MetadataRequestTopic().setName("foo"),
+                        new MetadataRequestData.MetadataRequestTopic().setName("bar")
+                )));
+        testMessageRoundTripsFromVersion(new MetadataRequestData().
+                setTopics(null).
+                setAllowAutoTopicCreation(true).
+                setIncludeClusterAuthorizedOperations(false).
+                setIncludeTopicAuthorizedOperations(false), (short) 1);
+        testMessageRoundTripsFromVersion(new MetadataRequestData().
+                setTopics(null).
+                setAllowAutoTopicCreation(false).
+                setIncludeClusterAuthorizedOperations(false).
+                setIncludeTopicAuthorizedOperations(false), (short) 4);
+    }
+
+    @Test
+    public void testHeartbeatVersions() throws Exception {
+        Supplier<HeartbeatRequestData> newRequest = () -> new HeartbeatRequestData()
+                .setGroupId("groupId")
+                .setMemberId("memberId")
+                .setGenerationId(15);
+        testMessageRoundTrips(newRequest.get());
+        testMessageRoundTrips(newRequest.get().setGroupInstanceId(null));
+        testMessageRoundTripsFromVersion(newRequest.get().setGroupInstanceId("instanceId"), (short) 3);
+    }
+
+    @Test
+    public void testJoinGroupVersions() throws Exception {
+        Supplier<JoinGroupRequestData> newRequest = () -> new JoinGroupRequestData()
+                .setGroupId("groupId")
+                .setMemberId("memberId")
+                .setProtocolType("consumer")
+                .setProtocols(new JoinGroupRequestData.JoinGroupRequestProtocolCollection())
+                .setSessionTimeoutMs(10000);
+        testMessageRoundTrips(newRequest.get());
+        testMessageRoundTripsFromVersion(newRequest.get().setRebalanceTimeoutMs(20000), (short) 1);
+        testMessageRoundTrips(newRequest.get().setGroupInstanceId(null));
+        testMessageRoundTripsFromVersion(newRequest.get().setGroupInstanceId("instanceId"), (short) 5);
+    }
+
+    @Test
+    public void testSyncGroupDefaultGroupInstanceId() throws Exception {
+        Supplier<SyncGroupRequestData> request = () -> new SyncGroupRequestData()
+                .setGroupId("groupId")
+                .setMemberId("memberId")
+                .setGenerationId(15)
+                .setAssignments(new ArrayList<>());
+        testMessageRoundTrips(request.get());
+        testMessageRoundTrips(request.get().setGroupInstanceId(null));
+        testMessageRoundTripsFromVersion(request.get().setGroupInstanceId("instanceId"), (short) 3);
+    }
+
+    @Test
+    public void testOffsetCommitDefaultGroupInstanceId() throws Exception {
+        testMessageRoundTrips(new OffsetCommitRequestData()
+                .setTopics(new ArrayList<>())
+                .setGroupId("groupId"), (short) 0);
```

Comment:
The last parameter seems not correct? Should this be a `testMessageRoundTripsFromVersion` call?

Reply:
Yes, it should be. Thanks.